### PR TITLE
Update build script to track unit-test failures

### PR DESCRIPTION
**Pull request by bprashan**
Signed-off-by: B <prashanth.natraj.b@intel.com>

### CODE REVIEW

File: cDevice_Build.sh [hunk 1]
```diff
@@ -55,8 +55,15 @@ cp -a ${WORKSPACE}/data/platform_hmac_key.bin ${WORKSPACE}/ecdsa384_c_sct_device
 cp -a ${WORKSPACE}/data/platform_iv.bin ${WORKSPACE}/ecdsa384_c_sct_device_bin/blob_backup
 
 echo " *****Running Unit Tests*********" 
-make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa ; make -j$(nproc)
-make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=cbc -DDA=ecdsa256 -DPK_ENC=ecdsa ; make -j$(nproc)
-make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh384 -DAES_MODE=ctr -DDA=ecdsa384 -DPK_ENC=ecdsa ; make -j$(nproc)
-make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh384 -DAES_MODE=cbc -DDA=ecdsa384 -DPK_ENC=ecdsa ; make -j$(nproc)
-make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa -DDA_FILE=pem ; make -j$(nproc)
+TEST_OUTPUT="unit-test-output.txt"
```
Comment (AdithyaBaglody) [inline cDevice_Build.sh:58]: we already create a file "/build/report.log"
why cant we reuse this?

Comment (trbehera) [reply cDevice_Build.sh:58]: Parsing 'build/report.log' for unit-test failures looks more complicated that parsing the output of 'cmake -Dunit-test=true'.

If no concerns with creation of unit-test-output.txt (if required, we can move this file inside build folder), we can go ahead with this approach.

Comment (AdithyaBaglody) [reply cDevice_Build.sh:58]: no i don't think it is tough, a simple awk script should be ok to handle it.
but its okay. just moving the new file inside build/ folder would be good.

File: cDevice_Build.sh [hunk 2]
```diff
@@ -55,8 +55,15 @@ cp -a ${WORKSPACE}/data/platform_hmac_key.bin ${WORKSPACE}/ecdsa384_c_sct_device
...
+rm -f $TEST_OUTPUT
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa ; make -j$(nproc) | tee -a $TEST_OUTPUT
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=cbc -DDA=ecdsa256 -DPK_ENC=ecdsa ; make -j$(nproc) | tee -a $TEST_OUTPUT
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh384 -DAES_MODE=ctr -DDA=ecdsa384 -DPK_ENC=ecdsa ; make -j$(nproc) | tee -a $TEST_OUTPUT
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh384 -DAES_MODE=cbc -DDA=ecdsa384 -DPK_ENC=ecdsa ; make -j$(nproc) | tee -a $TEST_OUTPUT
+make pristine || true; cmake -Dunit-test=true -DHTTPPROXY=true -DBUILD=release -DKEX=ecdh -DAES_MODE=ctr -DDA=ecdsa256 -DPK_ENC=ecdsa -DDA_FILE=pem ; make -j$(nproc) | tee -a $TEST_OUTPUT
+
+fail_list=$(cat $TEST_OUTPUT | grep -e "^Tests Failed" | cut -d":" -f2)
+fail_count=$(echo $fail_list | awk '{for(i=1;i<=NF;i++) sum=sum+$i;print sum}')
```
Comment (AdithyaBaglody) [inline cDevice_Build.sh:67]: unnecessary lines of code.
awk '/Tests Failed  :/ {split($0,a,": "); count+=a[2]} END{print count}' unit-test-output.txt

**Pull request merged**